src/Nova/RecurringSchedule.php

```diff
@@ -0,0 +1,121 @@
+            'Admin',
+            'Owner',
+            'Accountant',
+            'Executive',
+            'Reservation Manager',
+            'Reservationist',
+        ])) {
+            return $query;
+        }
+
+        return $request->withOrdering($request->withFilters(
+            $query->select('recurring_schedules.*')
+                ->join('rooms', 'rooms.id', '=', 'recurring_schedules.room_id')
+        ));
+    }
+
+    public static $group = 'Operations Scheduling';
+
+    public static $perPageViaRelationship = 30;
+
+    public function fieldsForIndex(NovaRequest $request)
+    {
+        return array_filter([
+            ID::make()->sortable(),
+            Text::make('Room', 'rooms.id', function () {
+                return $this->room->name;
+            })->sortable(),
+            Select::make('Day')->options([
+                1 => 'Monday',
+                2 => 'Tuesday',
+                3 => 'Wednesday',
+                4 => 'Thursday',
+                5 => 'Friday',
+                6 => 'Saturday',
+                7 => 'Sunday',
+            ])->displayUsingLabels()->sortable(),
+            TimeField::make('Time')->withTwelveHourTime()->sortable(),
+            Text::make('Rate', 'rate.id', function () {
+                return $this->rate->name;
+            })->sortable(),
+        ]);
+    }
+
+    public function fields(Request $request)
+    {
+        return array_filter([
+            nova('room') ? BelongsTo::make('Room', 'room', nova('room'))->required() : null,
+            Select::make('Day of the Week', 'day')->options([
+                1 => 'Monday',
+                2 => 'Tuesday',
+                3 => 'Wednesday',
+                4 => 'Thursday',
+                5 => 'Friday',
+                6 => 'Saturday',
+                7 => 'Sunday',
+            ])->required(),
+            TimeField::make('Time')->withTwelveHourTime()->required(),
+            nova('rate') ? BelongsTo::make('Rate', 'rate', nova('rate'))->hideWhenCreating()->nullable() : null,
+
+            new Panel('Optional Date Fields', $this->dateFields()),
+
+            new Panel('Data Fields', $this->dataFields()),
+        ]);
+    }
+
+    protected function dateFields()
+    {
+        return [
+            Date::make('Valid from')->nullable(),
+            Date::make('Valid until', 'expires_at')->nullable(),
+        ];
+    }
+
+    protected function dataFields(): array
+    {
+        return array_filter([
+            ID::make(),
+            nova('user') ? BelongsTo::make('Created By', 'creator', nova('user'))->exceptOnForms() : null,
+            DateTime::make('Created At')->exceptOnForms(),
+            nova('user') ? BelongsTo::make('Updated By', 'updater', nova('user'))->exceptOnForms() : null,
+            DateTime::make('Updated At')->exceptOnForms(),
+        ]);
+    }
+
+    public function filters(Request $request)
+    {
+        return [
+            new Filters\RoomLocation,
```

Comment:
@drewroberts can you let me know what packages contain these filters so I can refactor them?

Reply:
https://github.com/tipoff/escape-room